### PR TITLE
Add DCO signoff instructions

### PR DESCRIPTION
**Pull request by uberskigeek**
This is to add instructions and tools for signing off on a developer certificate of origin. 
Signed-off-by: Mike Andrasak <andrasak@us.ibm.com>

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+##Contributing
+All boost repositories require DCO signing.
```
Comment (scottkurz) [inline docs/CONTRIBUTING.md]: We just have one though... All => "the"

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+By adding a signed off line to your commit, you sign off this [Developer Certificate of Origin](https://developercertificate.org).
+You can add the signed off line to your commit by adding the -s flag to your commit or have it
```
Comment (scottkurz) [inline docs/CONTRIBUTING.md]: Can we just link to this prepare-commit-msg somewhere as an idea instead of owning it?

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+##Contributing
```
Comment (scottkurz) [inline docs/CONTRIBUTING.md]: Let's surface up a bit to ./CONTRIBUTING.md so it's more obvious from the top-level... AND make sure we have a link to it from the top-level README.md

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (uberskigeek) [reply docs/CONTRIBUTING.md]: ok

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (uberskigeek) [reply docs/CONTRIBUTING.md]: ok

**Pull request merged**